### DISCUSSION TITLE
Valkey

### DISCUSSION THREAD

**Discussion by mrbluecoat** [redis/go-redis#3149] [Q&A] [answered]
Is Valkey a drop-in replacement for Redis or are changes needed within go-redis?

**Comment by mortensi** [comment, marked as answer]
We test Redis libraries against all Redis products (Redis Community Edition, Redis Software, and Redis Cloud), so they are not guaranteed to work with forks. For example, Redis client libraries already support search, vector search, JSON documents, probabilistic data structures, and time series, which Redis forks or imitators haven't implemented or partially, to the best of my knowledge. So the answer to your question is no, ValKey is not a drop-in replacement.
